source/features/add-toggle-files-button.js

```diff
@@ -19,7 +24,11 @@ function addButton() {
 }
 
 export default function () {
-	if (!select.exists('.rgh-toggle-files')) {
-		addButton();
-	}
+	observeEl(commitTease, async () => {
+		await elementReady(commitTeaseTarget);
```

Comment:
I don’t think we need this. We’re already observing the container. If it doesn’t exist now, it will exist at the next childList change, right?

Reply:
Observing the container doesn't just cut it. We still need to wait for `.float-right` element to be available in DOM (I tried with and without `elementReady`, also tried rAF and double rAF too)

Reply:
If it's not enough, then it's not observing the right container. The point of `observeEl` is to wait for the browser to tell us when new elements appear, so combining it with `elementReady` isn't right

Reply:
Process:
* Page loads (with somewhat empty `.commit-tease` element
* Ajax request to recent commit information
* Replace content of `.commit-tease` with the new HTML

----------------

`observeEl` watches for `.commit-tease` to change (when it's content is replaced), then it takes some time (unknown) for the `.commit-tease .float-right` element to be available in DOM (i.e., uses `await elementReady` for each change in `.commit-tease`).

Reply:
`.commit-tease` is the wrong element to observe.

```
view-source:https://github.com/sindresorhus/refined-github
```

It's a `include-fragment` that is removed, so you can't observe it directly because the event we're waiting for is when `include-element` itself is removed. You probably have to observe `.repository-content, which is its parent.

Reply:
> You probably have to observe .repository-content, which is its parent

That worked (atleast for me)!